Assets/MRTK/Core/Providers/Hands/BaseHandVisualizer.cs

```diff
@@ -104,22 +105,30 @@ void IMixedRealityHandJointHandler.OnHandJointsUpdated(InputEventData<IDictionar
                 if (handTrackingProfile != null && !handTrackingProfile.EnableHandJointVisualization)
                 {
                     // clear existing joint GameObjects / meshes
-                    foreach (var joint in joints)
+                    foreach (Transform joint in jointsArray)
                     {
-                        Destroy(joint.Value.gameObject);
+                        if (joint != null)
+                        {
+                            Destroy(joint.gameObject);
+                        }
                     }
 
-                    joints.Clear();
                     return;
                 }
 
-                foreach (TrackedHandJoint handJoint in eventData.InputData.Keys)
+                for (int i = 0; i < ArticulatedHandPose.JointCount; i++)
                 {
-                    Transform jointTransform;
-                    if (joints.TryGetValue(handJoint, out jointTransform))
+                    TrackedHandJoint handJoint = (TrackedHandJoint)i;
+
+                    if (handJoint == TrackedHandJoint.None) { continue; }
```

Comment:
We weren't previously, but that's because we don't put `None` in the eventData's dictionary to begin with. Now that we're iterating through values independent of that dictionary, we have to filter out some enum values that we don't care about. Actually though...you bring up a good point and an optimization here: why not just start the `for` loop from `0`? 🤔